addons/mrp/models/mrp_workorder.py

```diff
@@ -520,6 +520,8 @@ def _plan_workorder(self, replan=False):
         best_finished_date = datetime.max
         vals = {}
         for workcenter in workcenters:
+            if not workcenter.resource_calendar_id:
+                continue
```

Comment:
You can raise an error directly saying a calendar is mandatory to not raise the following one (the message is not the good one for that scenario) 
https://github.com/odoo/odoo/pull/110548/files#diff-fac872ffb03b811c4976eb2e52991ec544265332df814d92cfda658a5b917423R545